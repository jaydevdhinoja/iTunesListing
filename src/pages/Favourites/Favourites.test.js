import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { renderWithContext, InitialState } from '../../tests/common';
import Favourites from '.';
import { screen } from '@testing-library/react';

describe('Page Listing', () => {
	test('renders component', () => {
		const { container } = renderWithContext(<Favourites />, { value: InitialState });

		expect(container).toMatchSnapshot();
	});

	test('renders component with no favouites albums', () => {
		const { queryAllByText } = renderWithContext(<Favourites />, {
			value: { ...InitialState, favourites: {...InitialState.favourites, albums: ['']}}
		});

		expect(queryAllByText('Black Pumas')).toHaveLength(0);
	});

	test('renders component with no favourite songs', () => {
		const { queryAllByText } = renderWithContext(<Favourites />, {
			value: { ...InitialState, favourites: {...InitialState.favourites, songs: [''] }}
		});

		expect(queryAllByText('Tim McGraw & Tyler Hubbard')).toHaveLength(0);
	});
	
	test('renders favourite albums', () => {
		const { getByText } = renderWithContext(<Favourites />, { value: InitialState });
		expect(getByText('Black Pumas')).toBeInTheDocument();
	});

	test('renders favourite songs', () => {
		const { getByText } = renderWithContext(<Favourites />, { value: InitialState });
		expect(getByText('Tim McGraw & Tyler Hubbard')).toBeInTheDocument();
	});
});
