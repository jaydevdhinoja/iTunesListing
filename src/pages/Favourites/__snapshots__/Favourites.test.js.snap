// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Listing renders component 1`] = `
<div>
  <div
    class="favourites-page"
  >
    <div
      class="container wrapper"
    >
      <section
        class="albums-grid container"
      >
        <div
          class="albums-grid__headline section-headline"
        >
          <h2>
            Favourite Albums
          </h2>
        </div>
        <ul
          class="albums-grid__list"
        >
          <li
            class="item-card"
          >
            <a
              href="/item/1524099463"
            >
              <div
                class="item-card__mobile-link"
              />
            </a>
            <div
              class="item-card__thumbnail"
              style="background-image: url(undefined);"
            >
              <img
                alt="Heart Icon"
                class="favourite-icon"
                src="heart-solid.svg"
              />
            </div>
            <h3
              class="item-card__title"
            >
              Black Pumas (Deluxe)
            </h3>
            <p
              class="item-card__artist"
            >
              Black Pumas
            </p>
            <div
              class="item-card__overlay"
            />
            <p
              class="item-card__overlay-number"
            >
              1
            </p>
            <a
              href="/albums/1524099463"
            >
              <span
                class="item-card__overlay-rect"
              >
                <p>
                  See albums
                </p>
              </span>
            </a>
            <span
              class="item-card__overlay-shadow"
            />
          </li>
        </ul>
      </section>
      <section
        class="albums-grid container"
      >
        <div
          class="albums-grid__headline section-headline"
        >
          <h2>
            Favourite Songs
          </h2>
        </div>
        <ul
          class="albums-grid__list"
        >
          <li
            class="item-card"
          >
            <a
              href="/item/1545178423"
            >
              <div
                class="item-card__mobile-link"
              />
            </a>
            <div
              class="item-card__thumbnail"
              style="background-image: url(undefined);"
            >
              <img
                alt="Heart Icon"
                class="favourite-icon"
                src="heart-solid.svg"
              />
            </div>
            <h3
              class="item-card__title"
            >
              Undivided
            </h3>
            <p
              class="item-card__artist"
            >
              Tim McGraw & Tyler Hubbard
            </p>
            <div
              class="item-card__overlay"
            />
            <p
              class="item-card__overlay-number"
            >
              1
            </p>
            <span
              class="item-card__overlay-shadow"
            />
          </li>
        </ul>
      </section>
    </div>
  </div>
</div>
`;
